bazel_tools/scala.bzl

```diff
@@ -62,6 +62,7 @@ common_scalacopts = version_specific.get(scala_major_version, []) + [
     "-unchecked",
     # warn if using deprecated stuff
     "-deprecation",
+    "-Wconf:cat=deprecation:wv,cat=feature:wv,cat=optimizer:wv",
```

Comment:
Can we add a comment on what this does?

Reply:
I'm removing it because `compatibility` doesn't understand it. Do we use a different version of Scala there?

Reply:
> I'm removing it because `compatibility` doesn't understand it. Do we use a different version of Scala there?

Not deliberately but I wouldn’t be surprised if we ended up getting the two out of sync somehow.

Reply:
Just checked. We do, of course. I'll bump it in a separate PR.

This line was "nice to have" for more verbose warnings anyway; it's not necessary.